pkg/innerring/contracts.go

```diff
@@ -121,11 +145,60 @@ func parseAlphabetContracts(cfg *viper.Viper, morph *client.Client) (alphabetCon
 	return alpha, nil
 }
 
-func parseContract(cfg *viper.Viper, morph *client.Client, cfgName, nnsName string) (res util.Uint160, err error) {
-	contractStr := cfg.GetString(cfgName)
-	if len(contractStr) == 0 {
-		return morph.NNSContractAddress(nnsName)
+type nnsContext struct {
+	context.Context
+
+	nnsDeployed bool
+}
+
+func parseContract(ctx *nnsContext, _logger *logger.Logger, cfg *viper.Viper, morph *client.Client, cfgName, nnsName string) (res util.Uint160, err error) {
+	if cfg.IsSet(cfgName) {
+		_logger.Debug("decoding configured contract...", zap.String("config key", cfgName))
+		return util.Uint160DecodeStringLE(cfg.GetString(cfgName))
+	}
+
+	const pollInterval = 5 * time.Second // no too long and not so frequently
```

Comment:
You can use time per block setting, you have it here (or can get from a client). It'd be more correct for a particular network (1s or 15s, there is a difference).